config/compiler/image.go

```diff
@@ -1,11 +1,29 @@
 package compiler
 
 import (
-	"github.com/docker/docker/reference"
+	"fmt"
+	"strings"
+
+	"github.com/docker/distribution/reference"
+)
+
+const (
+	// defaultTag defines the default tag used when performing images related actions and no tag or digest is specified
+	defaultTag = "latest"
+	// defaultHostname is the default built-in hostname
+	defaultHostname = "docker.io"
+	// legacyDefaultHostname is automatically converted to DefaultHostname
+	legacyDefaultHostname = "index.docker.io"
+	// defaultRepoPrefix is the prefix used for default repositories in default host
+	defaultRepoPrefix = "library"
 )
 
 // trimImage returns the short image name without tag.
 func trimImage(name string) string {
+	name = strings.TrimPrefix(name, legacyDefaultHostname+"/")
+	name = strings.TrimPrefix(name, defaultHostname+"/")
+	name = strings.TrimPrefix(name, defaultRepoPrefix+"/")
```

Comment:
it looks like there is a `reference.Path` function that can be used to trim the path from the image https://godoc.org/github.com/docker/distribution/reference#Path